source/grammar/tests/invalid/statements/declarations.qasm

```diff
@@ -31,6 +31,12 @@ complex[creg] myvar;
 complex[qreg[8]] myvar;
 complex[creg[8]] myvar;
 
+// Bad array specifiers.
+array myvar;
+array[8] myvar;
+array[int, 8] myvar;  // (no width for the integer)
```

Comment:
Why does this fail? Seems like this should work taking the compiler's definition of `int`.

Reply:
At the moment the ANTLR grammar doesn't permit the statement `int x;` with no width, so this is just consistent with that.  I don't remember whether the idea of "generic/default int" is present in the spec at the moment, but if it is, then we should update this in a separate PR - there's a bunch more of the grammar that would need updating too.  I think it's not there yet, though it's something we've been discussing.

Reply:
This seems to be something we neglected to translate over from the paper (see section 4.4 of https://arxiv.org/abs/2104.14722)